File: UNP/ref/C/Funcoes/multiplica_recursivo.c
#include <stdio.h>

int multiplica(int num1, int num2);

int main(){
    int i,j;
    for (i = 2;i<=12;i++){
        printf("Tabuada de %d\n=====================\n", i);
        for (j = 1;j<=12;j++){
            printf("%d x %d = %d\n", i,j, multiplica(i,j));
        }
        printf("=====================\n");
    }
    return 0;
}

int multiplica(int num1, int num2){
    if (num1 == 0 || num2 == 0) {        
        return 0;
    }
    //caso base, onde a recursão para:
    else if (num2 == 1) {        
        return num1;
    }
    //multiplicando através da soma com recursividade:
    else {        
        return (num1 + multiplica(num1,num2 - 1));        
    }
}

File: UNP/ref/Java/#ED2-Java-BKP/recursividade/MultiplicaRecursivo.java
package recursividade;
public class MultiplicaRecursivo {

    public static void main(String[] args) {
        for (int i = 2; i <= 12; i++) {
            System.out.printf("Tabuada de %d\n===============\n", i);
            for (int j = 1; j <= 10; j++) {
                System.out.printf("%-2d x %-2d = %-3d\n", i, j, multiplicaRecursivo(i,j));
            }
            System.out.println("=======================");
        }
    }
    public static int multiplicaRecursivo(int a, int b){
        if (a==0||b==0) return 0;
        else if (b==1) {
            return a;
        }
        else{
            return (a + multiplicaRecursivo(a, b-1));
        }
    }

}
